app/src/main/java/net/bible/service/cloudsync/nextcloud/NextCloudAdapter.kt

```diff
@@ -0,0 +1,230 @@
+ *
+ * AndBible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with AndBible.
+ * If not, see http://www.gnu.org/licenses/.
+ */
+
+package net.bible.service.cloudsync.nextcloud
+
+import android.net.Uri
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.OwnCloudClientFactory
+import com.owncloud.android.lib.common.OwnCloudCredentialsFactory
+import com.owncloud.android.lib.common.operations.OnRemoteOperationListener
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.files.*
+import com.owncloud.android.lib.resources.files.model.RemoteFile
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import net.bible.android.database.SyncConfiguration
+import net.bible.android.view.activity.base.ActivityBase
+import net.bible.service.cloudsync.CloudAdapter
+import net.bible.service.cloudsync.CloudFile
+import net.bible.service.cloudsync.GZIP_MIMETYPE
+import net.bible.service.cloudsync.SyncableDatabaseAccessor
+import net.bible.service.cloudsync.TAG
+import net.bible.service.common.CommonUtils
+import net.bible.service.common.asyncMap
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.OutputStream
+import java.util.UUID
+import kotlin.collections.List
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+const val FOLDER_MIMETYPE = "DIR"
+const val NEXTCLOUD_SECRET_FILE_NAME_KEY = "nextCloudSecretFile"
+
+class NextCloudAdapter(
+    private val serverUrl: String?,
+    private val username: String?,
+    private val password: String?
+) : CloudAdapter {
+    private var _client: OwnCloudClient? = null
+    private val client get() = _client!!
+
+    override val signedIn: Boolean get() = _client != null
+
+    override suspend fun signIn(activity: ActivityBase): Boolean = withContext(Dispatchers.IO) {
+        try {
+            val serverUri = Uri.parse(serverUrl)
+            _client = OwnCloudClientFactory.createOwnCloudClient(serverUri, activity, true).apply {
+                credentials = OwnCloudCredentialsFactory.newBasicCredentials(username, password)
+                userId = username
+            }
+            return@withContext true
+        } catch (e: Exception) {
+            Log.e(TAG, "Login to NextCloud failed", e)
+            _client = null
+            return@withContext false
+        }
+    }
+
+    override suspend fun signOut() {
+        _client = null
```

Comment:
sign out should also remove configurations (getConfigs()). After signing out it must be again fresh start.

Reply:
actually that is taken care of in CloudSync.signOut()